apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/adapter/JavaxServletApiAdapter.java

```diff
@@ -43,7 +43,17 @@ public class JavaxServletApiAdapter implements ServletApiAdapter<HttpServletRequ
 
     private static final JavaxServletApiAdapter INSTANCE = new JavaxServletApiAdapter();
 
+    private final boolean ServletContext_getClassLoader_available;
+
     private JavaxServletApiAdapter() {
+        boolean probeResult;
+        try {
+            ServletContext.class.getMethod("getClassLoader");
+            probeResult = true;
+        } catch (NoSuchMethodException e) {
+            probeResult = false;
+        }
+        ServletContext_getClassLoader_available = probeResult;
```

Comment:
I think it's cleaner to have
`ServletContext_getClassLoader_available = canExecuteServletContextgetClassLoader()`
as the constructor, but that's my preference, will approve as is